### ISSUE TITLE
DDC ModuleBuilder with fine strategy doesn't handle conditional imports

### ISSUE THREAD

**Issue by nickclmb** [dart-lang/build#1752]
When ModuleBuilder runs with fine strategy (as example packages with circular dependencies) then .module files are missing dependencies from conditional imports:

```
import 'A.dart'                            // this uri present in deps
if (dart.library.io) 'A_io.dart'           // <-- missing
if (dart.library.html) 'A_html.dart';      // <-- missing
```

So as a result, DDC fails with error "'A_html.dart' not found."

As i can see the LibraryElement from resolver doesn't contain any info about import configurations, but ModuleLibrary output already has all dependencies.
So is it possible to use ModuleLibrary to build modules in Fine strategy instead of using resolver?

**Comment by jakemac53** [comment]
Not using a Resolver here should also improve build times for projects that don't otherwise require a resolver.

**Comment by natebosch** [comment]
I had an experimental branch using `ModuleLibrary` for the fine strategy and hit some roadblocks - it got a little complex because it can't be done in the number of phases we're using today. I think it could be done with an additional intermediate output though.

**Comment by jakemac53** [comment]
The fix is published now with build_modules 0.3.2
